src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.cs

```diff
@@ -1425,18 +1425,18 @@ private static void ThrowNullLowHighInclusive<T>(T? lowInclusive, T? highInclusi
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static unsafe bool SequenceEqual<T>(this Span<T> span, ReadOnlySpan<T> other) where T : IEquatable<T>?
         {
-            int length = span.Length;
-
+            nuint length = (nuint)span.Length;
+            nuint otherLength = (nuint)other.Length;
```

Comment:
avoid sign extension

```suggestion
            nuint length = (uint)span.Length;
            nuint otherLength = (uint)other.Length;
```